docs/src/index.md

```diff
@@ -1,24 +1,106 @@
-* [`Scalar`](@ref)
-* [`One`](@ref)
-* [`Zero`](@ref)
+* an easy-to-use interface for constructing geometric algebra objects,
 
-## Operations
+* intuitive notation for operations on and between geometric algebra objects, and
+
+* behavior (including limitations) analogous to standard numerical linear algebra
+  libraries.
+
+## Overview
+
+Geometric algebra extends classical linear algebra (of inner product spaces over the real
+numbers ``\mathbb{R}``) by including the higher-dimensional siblings of the vector (a
+1-dimensional object) and defining algebraic operations that augment the basic operations
+on inner product spaces (i.e., vector addition, scalar multiplication, and inner product).
+With these extensions, geometric algebra makes it possible to use simple algebraic
+expressions to represent geometric objects (e.g., subspaces) and operations (e.g.,
+projection, computing the orthogonal complement).
+
+The key mathematical object introduced in geometric algebra is the _blade_. Geometrically,
+blades represent two concepts (1) subspaces of ``\mathbb{R}^n`` and (2) ``k``-dimensional
+hypervolumes (with dimension less than or equal to ``n`` and arbitrary shape). The
+_exterior product_ (``\wedge``), which combines lower-grade blades into higher-grade blades
+(or zero), is an example of an algebraic operation with geometric meaning. For instance,
+it enables any blade ``B`` to be expressed _algebraically_ in terms of a basis
+``b_1, \ldots b_k`` for the subspace represented by the blade:
+
+```math
+B = b_1 \wedge \cdots \wedge b_k.
+```
+
+The GeometricAlgebra package implements types and functions that make it straightforward
+to do _computational_ geometric algebra calculations.
+
+## Getting Started
+
+Install the GeometricAlgebra package via the Pkg REPL. That's it!
+
+```julia
+pkg> add GeometricAlgebra  # Press ']' to enter the Pkg REPL mode.
+```
+
+## Summary of Types
+
+GeometricAlgebra defines concrete types for the following hierarchy of mathematical objects.
+
+* multivector: [`Multivector`](@ref)
+  * blade: [`Blade`](@ref)
+    * scalar (``0``-blade) : [`Scalar`](@ref)
+      * multiplicative identity: [`One`](@ref)
+      * additive identity: [`Zero`](@ref)
+    * pseudoscalar (``n``-blade): [`Pseudoscalar`](@ref)
+
+## Summary of Operations
 
 ### Unary Operations
 
-* [`inverse`](@ref), [`-`](@ref GeometricAlgebra.inverse)
-* [`reverse`](@ref)
-* [`dual`](@ref)
+* additive inverse: [`inverse`](@ref), [`-`](@ref GeometricAlgebra.inverse)
+* multiplicative inverse: [`reciprocal`](@ref)
+* reverse: [`reverse`](@ref)
+* dual: [`dual`](@ref)
 
 ### Binary Operations
 
-* TODO
+* addition/subtraction: [`+`](@ref Base.:(+)), [`-`](@ref)
+* multiplication/division (geometric product): [`*`](@ref), [`/`](@ref)
+* exterior product: [`wedge`](@ref), [`∧`](@ref)
+* inner product (left contraction): [`contract_left`](@ref),
+  [`<`](@ref GeometricAlgebra.contract_left), [`\cdot`](@ref GeometricAlgebra.contract_left)
```

Comment:
I don't see any implementation of the `\cdot` alias, should we implement it or should we remove it from the docs?

Sidenote: the `\cdot` does not render for me when I build the docs myself, should it render LaTeX or is it expected to show up as plain text?

Reply:
I changed the `\cdot` entry to the following.

```
  [`dot`](@ref GeometricAlgebra.contract_left),
  [`⋅`](@ref GeometricAlgebra.contract_left)
```

We will probably need to update that in a near-future release to sync up with the operator proposal from the general GeometricAlgebra.jl group.

Reply:
When I only call `using GeometricAlgebra` without `using LinearAlgebra`, I'm able to call `dot` but not its alias `⋅` (they both work if you call `using LinearAlgebra`). This seems confusing to me, should we keep it like this for now anyways?
```
julia> using GeometricAlgebra

julia> A = Zero()
Zero{Float64}()

julia> B = One()
One{Float64}()

julia> dot(A, B)
Zero{Float64}()

julia> A ⋅ B
ERROR: UndefVarError: ⋅ not defined
Stacktrace:
 [1] top-level scope
   @ REPL[5]:1
```

Reply:
@wgvozdjak Yes. I agree that this is confusing. Can you create a separate GitHub Issue to track this? This issue is related to the one that we have been discussing in the JuliaGeometricAlgebra/GeometricAlgebra.jl project.

For the initial version of the package docs, I have removed the reference for \cdot.

Reply:
Sounds good, I created issue #40.